src/APL/firestore-apl.test.ts

```diff
@@ -0,0 +1,72 @@
+import { CollectionReference, Firestore } from "@google-cloud/firestore";
+import { describe, expect, it } from "vitest";
+
+import { AuthData } from "./apl";
+import { FirestoreAPL } from "./firestore-apl";
+
+let firestore: Firestore | null = null;
+
+try {
+  firestore = new Firestore({
+    // @ts-ignore ignore import.meta, TODO
+    projectId: import.meta.env.TEST_FIRESTORE_PROJECT_ID,
+    credentials: {
+      // @ts-ignore ignore import.meta, TODO
+      private_key: import.meta.env.TEST_FIRESTORE_PRIVATE_KEY,
+      // @ts-ignore ignore import.meta, TODO
+      client_email: import.meta.env.TEST_FIRESTORE_CLIENT_EMAIL,
+    },
+  });
+} catch (e) {
+  // Omit test
+}
+
+describe.runIf(Boolean(firestore))("Firestore APL", async () => {
```

Comment:
I dont want to introduce complexty of writing e2e tests, and also Firestore is specific and hard to mock. Thats way these tests are opt in. If values are provided, they work. We should probably add them to CICD but don't require them locally

Reply:
WDYT bout testing it with the Polly.js? 
- no need to create custom mocks since the real communication with the API is recorded and replayed
- updating the recording would be only required if a query to the Firestore API would change

Reply:
I will add separate issue for that, don't want to block it because of way we test (not sure how much time setup of polly will take)